v3.6.1 coredumps on close

Hi,

Aiming to update the NetBSD package, I did a build of simp-3.6.1 from source within pkgsrc framework. 
The resulting binary coredumps when closing the application. This is consistent and happens every time.
`gdb` backtrace doesn't look particularly useful to me but, in case it makes sense to you, here it is.

```
gdb simp simp.core
GNU gdb (GDB) 13.2
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64--netbsd".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from simp...
(No debugging symbols found in simp)
[New process 3260]
[New process 2419]
[New process 2418]
[New process 3452]
Core was generated by `simp'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007856c70ae407 in ?? ()
[Current thread is 1 (process 3260)]
(gdb) bt
#0  0x00007856c70ae407 in ?? ()
#1  0x00007856c875821f in ?? ()
#2  0x0000000001c703b8 in ?? ()
#3  0x0000000001c703b8 in ?? ()
#4  0x0000000000000000 in ?? ()
```

Additional information:
```
~> uname -rsv
NetBSD 10.99.10 NetBSD 10.99.10 (GENERIC) #0: Sat Jun  1 15:54:40 UTC 2024  mkrepro@mkrepro.NetBSD.org:/usr/src/sys/arch/amd64/compile/GENERIC
~> rustc --version
rustc 1.78.0 (9b00956e5 2024-04-29) (built from a source tarball)
```

If you don't mind a separate question, I've been having issues with `Ctrl+O`or, File -> Open not doing anything. Launching it from the command line with "path to file" works fine but, the major issue is that the same happens when trying to save changes, File -> Save as doesn't open the Gtk dialogue, making it impossible to save changes.

This behavior was unfortunately already present in v3.5.3. I know it used to work fine prior to that, just can't remember at which exact version this stopped to work.

Thanks for any insights.

As for downgrading `rfd`, it doesn't seem like I'm allowed to do that,
```
error: failed to run custom build command for `simp v3.6.1 (/home/pin/simp)`

Caused by:
  process didn't exit successfully: `/home/pin/simp/target/release/build/simp-14689108e1328d47/build-script-build` (exit status: 101)
  --- stdout
  cargo:rustc-env=GIT_HASH=cd5d8296f9acc44113282525087381aa246652af

  cargo:rerun-if-changed=Cargo.toml

  --- stderr
  2024-06-10 21:01:48.518494225 +00:00:00 [WARN] the definition for crate/cratesio/async-signal/0.2.8 has not been harvested
  2024-06-10 21:01:48.889761682 +00:00:00 [WARN] the definition for crate/cratesio/cc/1.0.99 has not been harvested
  2024-06-10 21:01:49.287125579 +00:00:00 [WARN] the definition for crate/cratesio/clap/4.5.7 has not been harvested
  2024-06-10 21:01:49.28825536 +00:00:00 [WARN] the definition for crate/cratesio/clap_builder/4.5.7 has not been harvested
  2024-06-10 21:01:49.289340753 +00:00:00 [WARN] the definition for crate/cratesio/clap_lex/0.7.1 has not been harvested
  2024-06-10 21:01:49.674365831 +00:00:00 [WARN] the definition for crate/cratesio/enumflags2/0.7.10 has not been harvested
  2024-06-10 21:01:49.674475224 +00:00:00 [WARN] the definition for crate/cratesio/enumflags2_derive/0.7.10 has not been harvested
  2024-06-10 21:01:50.680113537 +00:00:00 [WARN] the definition for crate/cratesio/icu_normalizer/1.5.0 has not been harvested
  2024-06-10 21:01:50.680227771 +00:00:00 [WARN] the definition for crate/cratesio/icu_normalizer_data/1.5.0 has not been harvested
  2024-06-10 21:01:50.680769702 +00:00:00 [WARN] the definition for crate/cratesio/idna/1.0.0 has not been harvested
  2024-06-10 21:01:52.786627821 +00:00:00 [WARN] the definition for crate/cratesio/regex-automata/0.4.7 has not been harvested
  2024-06-10 21:01:52.786735377 +00:00:00 [WARN] the definition for crate/cratesio/regex-syntax/0.8.4 has not been harvested
  2024-06-10 21:01:52.786827197 +00:00:00 [WARN] the definition for crate/cratesio/regex/1.10.5 has not been harvested
  2024-06-10 21:01:53.698749845 +00:00:00 [WARN] the definition for crate/cratesio/toml/0.8.14 has not been harvested
  2024-06-10 21:01:53.699601155 +00:00:00 [WARN] the definition for crate/cratesio/toml_edit/0.22.14 has not been harvested
  2024-06-10 21:01:53.982607127 +00:00:00 [WARN] the definition for crate/cratesio/unicode-width/0.1.13 has not been harvested
  2024-06-10 21:01:53.983965989 +00:00:00 [WARN] the definition for crate/cratesio/url/2.5.1 has not been harvested
  2024-06-10 21:01:54.977866602 +00:00:00 [WARN] the definition for crate/cratesio/windows-result/0.1.2 has not been harvested
  2024-06-10 21:01:55.618665334 +00:00:00 [WARN] the definition for crate/cratesio/winnow/0.6.13 has not been harvested
  2024-06-10 21:01:55.898880389 +00:00:00 [WARN] the definition for crate/cratesio/xdg-home/1.2.0 has not been harvested
  2024-06-10 21:01:55.900102021 +00:00:00 [WARN] the definition for crate/cratesio/xkeysym/0.2.1 has not been harvested
  2024-06-10 21:02:03.857792422 +00:00:00 [WARN] LicenseRef-UFL-1.0 has no license file for crate 'epaint 0.27.2'
  error: failed to satisfy license requirements
     ┌─ /home/pin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/icu_collections-1.5.0/Cargo.toml:32:12
     │
  32 │ license = "Unicode-3.0"
     │            -----------
[...]
  error: failed to satisfy license requirements
     ┌─ /home/pin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/zerovec-derive-0.10.2/Cargo.toml:32:12
     │
  32 │ license = "Unicode-3.0"
     │            -----------

  2024-06-10 21:02:03.87405731 +00:00:00 [ERROR] encountered 19 errors resolving licenses, unable to generate output
  thread 'main' panicked at build.rs:58:9:
  assertion failed: exit_status.success()
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

--- a/src/main.rs
+++ b/src/main.rs
@@ -34,7 +34,7 @@ mod cli;
 mod image_io;
 
 #[derive(Serialize, Deserialize, Debug, Default)]
-pub struct Config {
+struct Config {
     maximized: bool,
     preferences: Preferences,
 }
@@ -211,7 +211,7 @@ impl<'a> WindowHandler<'a> {
 }
 
 impl WindowHandler<'_> {
-    pub fn main_loop(mut self) -> Config {
+    pub fn main_loop(mut self) {
         let WindowHandler {
             event_loop,
             mut app,
@@ -221,9 +221,6 @@ impl WindowHandler<'_> {
             ref mut egui_shapes,
             mut wgpu,
         } = self;
-
-        let mut config = Config::default();
-        let config_ref = &mut config;
 
         let _ = event_loop.run(move |event, event_loop| match event {
             Event::Resumed => wgpu.window.set_visible(true),
@@ -407,17 +404,16 @@ impl WindowHandler<'_> {
             },
             Event::LoopExiting => {
                 wgpu.window.set_visible(false);
-                let maximized = wgpu.window.is_maximized();
-                *config_ref = Config {
-                    maximized,
+                let data = Config {
+                    maximized: wgpu.window.is_maximized(),
                     preferences: PREFERENCES.lock().unwrap().clone(),
                 };
+                confy::store("simp", None, data).unwrap();
+                std::process::exit(0);
             }
             Event::UserEvent(mut event) => app.handle_user_event(&wgpu, &mut event),
             _ => (),
         });
-
-        config
     }
 }
 
@@ -457,9 +453,5 @@ fn main() {
             .app
             .queue(Op::LoadPath(PathBuf::from(path), true))
     }
-
-    let config = window_handler.main_loop();
-    if let Err(err) = confy::store("simp", None, config) {
-        eprintln!("{err}");
-    };
+    window_handler.main_loop();
 }

